File: part-05/juggler-starter/src/main/java/workshop/spring/beans/JugglerConfig.java
package workshop.spring.beans;

import org.springframework.boot.autoconfigure.AutoConfiguration;
import org.springframework.context.annotation.Bean;
import workshop.performer.Performer;

@AutoConfiguration
public class JugglerConfig {

    @Bean
    public Performer juggler() {
        return new Juggler();
    }

}

File: party-application/juggler-starter/src/main/kotlin/de/felixroske/spring/JugglerConfig.kt
package de.felixroske.spring

import de.felixroske.performer.Juggler
import de.felixroske.performer.Performer
import org.springframework.boot.autoconfigure.AutoConfiguration
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@AutoConfiguration
class JugglerConfig(val props: JugglerProperties) {
    @Bean
    open fun performer(): Performer = Juggler(props.item, props.num)
}